.github/actions/setup-job/action.yml

```diff
@@ -1,19 +1,36 @@
 name: Setup Job
 description: Common setup for all jobs
+inputs:
+    node-version:
+        description: 'The Node.js version to use'
+        required: false
+        default: '20'
 runs:
     using: 'composite'
     steps:
         - name: Enable Corepack
           shell: bash
           run: corepack enable
 
-        - name: Setup Node 20
+        - name: Setup Node LTS version
           uses: actions/setup-node@v4
           with:
-              node-version: '20'
+              node-version: ${{ inputs.node-version }}
               cache: 'yarn'
               registry-url: 'https://registry.npmjs.org'
 
+        ## --- YARN CACHE --- ##
+        - name: Check for cached dependencies
```

Comment:
Speeds the install up

Reply:
i could cry for this addition i love it so much